### ISSUE TITLE
Weatherlink Pressure Units

### ISSUE THREAD

**Issue by raf10gh** [siku2/hass-weatherlink#24]
Hi Siku2,

I see my pressure is displayed in PSI. I want it in inHg and tried changing in the Unit of Measure drop down to inHg and click Update but it doesn't save the changes.

![Pressure](https://user-images.githubusercontent.com/73025561/180605507-3ca29251-b679-47e4-ab88-41677c1c8802.png)

![Pressure_psi](https://user-images.githubusercontent.com/73025561/180576775-969849b8-a24b-4f67-b665-344a09a5dc82.png)

I see that under the Weatherlink device it is set to inHg.

![Pressure_inHG](https://user-images.githubusercontent.com/73025561/180576956-046d491b-4750-42e1-9da1-1358ded0ab7f.png)

Not sure if I'm doing something wrong or if it is a bug.

Thanks

Rick

**Comment by siku2** [comment]
The reason for this is that up until recently Hass didn't support switching units like that so this integration included its own system using the integration options.

**Comment by raf10gh** [comment]
Ok so is there a way to change it to inhG? In the Weatherlink screenshot the pressure units is set to inHg but it still displays in PSI.

**Comment by raf10gh** [comment]
Disregard I found it.

Thanks

**Comment by ppetru** [comment]
> The reason for this is that up until recently Hass didn't support switching units like that so this integration included its own system using the integration options.

Would you be willing to switch to the home assistant unit conversion system? I might try to help when I find some time.

**Comment by siku2** [comment]
> Would you be willing to switch to the home assistant unit conversion system? I might try to help when I find some time.

absolutely, I was never a fan of rolling my own in the first place. I was planning to remove it the next time I made any changes to the code.
